helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AbstractSchemaBuilder.java

```diff
@@ -250,6 +291,26 @@ public Schema build() {
         return result;
     }
 
+    private Schema inheritNamespace(Schema schema) {
+        if (!_inheritNamespace) {
+            return schema;
+        }
+
+        SchemaBuilder schemaBuilder = _adapter.newSchemaBuilder(schema).setInheritNamespace(_inheritNamespace);
+
+        //NamedSchema inherits parent's namespace only if it does not have namespace
+        Schema.Type schemaType = schema.getType();
+        if (schemaType == Schema.Type.FIXED || schemaType == Schema.Type.ENUM || schemaType == Schema.Type.RECORD) {
```

Comment:
we have a set of NAMED_TYPES somewhere you could use instead of this comparison chain

Reply:
changed to use NAMED_TYPES.